### PR TITLE
Fix enum example

### LINKED ISSUE #569: Rule 112 example in conflict with 240
The example for [rule 112](https://opensource.zalando.com/restful-api-guidelines/#112) is in conflict with rule [240](https://opensource.zalando.com/restful-api-guidelines/#240).

Additionally rule 240 is highly relevant for 112 and should therefore be referenced from 112.

### PR DESCRIPTION
**Pull request by vadeg**
* Update enum values example to match API guideline requirements.
* Add a link to rule 240 from rule 112

Fixes #569

### CODE REVIEW

**Comment by tfrauenstein** [comment]
👍

**Comment by vadeg** [comment]
@tkrop @whiskeysierra @ePaul could you please bump it?

File: chapters/compatibility.adoc [hunk 1]
```diff
@@ -199,14 +199,15 @@ follows:
 deliver_methods:
```
Comment (whiskeysierra) [inline chapters/compatibility.adoc]:
```suggestion
delivery_methods:
```

While we're at it.

File: chapters/compatibility.adoc [hunk 2]
```diff
@@ -196,17 +196,18 @@ follows:
...
-    - email
+    - PARCEL
+    - LETTER
+    - EMAIL
 ----
 
 *Note:* {x-extensible-enum} is not JSON Schema conform but will be ignored by
 most tools.
 
+See <<240>> about enum values formatting constraints.
```
Comment (whiskeysierra) [inline chapters/compatibility.adoc]:
```suggestion
See <<240>> about enum value naming conventions.
```

**Comment by tkrop** [comment]
:+1:

**Comment by vadeg** [comment]
👍

**Pull request merged**